examples/04_pong/audio.rs

```diff
@@ -1,56 +1,57 @@
-use futures::prelude::*;
 
 pub struct Sounds {
-    pub score_sfx: Source,
-    pub bounce_sfx: Source,
+    pub score_sfx: SourceHandle,
+    pub bounce_sfx: SourceHandle,
 }
 
 /// Loads an ogg audio track.
-fn load_audio_track(loader: &mut Loader, file: &str) -> Source {
+fn load_audio_track(loader: &Loader, world: &World, file: &str) -> SourceHandle {
     use amethyst::audio::OggFormat;
 
-    loader
-        .load_from(file, OggFormat, "assets")
-        .wait()
-        .expect(&format!("Could not load audio track: '{}'", file))
+    loader.load(
+        file,
+        OggFormat,
+        (),
+        &mut Progress::new(),
+        &world.read_resource(),
+    )
 }
 
 /// Initialise audio in the world. This includes the background track and the
 /// sound effects.
 pub fn initialise_audio(world: &mut World) {
     use {AUDIO_BOUNCE, AUDIO_MUSIC, AUDIO_SCORE};
-    use amethyst::audio::output::Output;
+    //use amethyst::audio::output::Output;
 
     let sound_effects = {
-        let mut loader = world.write_resource::<Loader>();
-        loader.register(AudioContext::new());
+        let loader = world.read_resource::<Loader>();
 
-        // Add a DJ if we have sound output and background music tracks.
-        if world.read_resource::<Option<Output>>().is_some() && AUDIO_MUSIC.len() > 0 {
-            let mut dj = world.write_resource::<Dj>();
-            dj.set_volume(0.25); // Music is a bit loud, reduce the volume.
-            let mut next_track_index = 0;
-
-            let music_tracks: Vec<_> = AUDIO_MUSIC
-                .iter()
-                .map(|file| load_audio_track(&mut loader, file))
-                .collect();
-
-            dj.set_picker(Box::new(move |ref mut dj| {
-                dj.append(&music_tracks[next_track_index])
-                    .expect("Decoder error occurred!");
-                next_track_index = (next_track_index + 1) % music_tracks.len();
-                true
-            }));
-        }
+        //        // Add a DJ if we have sound output and background music tracks.
+        //        if world.read_resource::<Option<Output>>().is_some() && AUDIO_MUSIC.len() > 0 {
+        //            let mut dj = world.write_resource::<Dj>();
+        //            dj.set_volume(0.25); // Music is a bit loud, reduce the volume.
+        //            let mut next_track_index = 0;
+        //
+        //            let music_tracks: Vec<_> = AUDIO_MUSIC
+        //                .iter()
+        //                .map(|file| load_audio_track(&loader, &world, file))
+        //                .collect();
+        //
+        //            dj.set_picker(Box::new(move |ref mut dj| {
+        //                dj.append(&music_tracks[next_track_index])
+        //                    .expect("Decoder error occurred!");
+        //                next_track_index = (next_track_index + 1) % music_tracks.len();
+        //                true
+        //            }));
+        //        }
```

Comment:
Should probably remove this outdated code.

Reply:
Let's remove it once we have music stuff where we want it to be.